llmfoundry/callbacks/eval_output_logging_callback.py

```diff
@@ -47,28 +47,31 @@ def eval_batch_end(self, state: State, logger: Logger) -> None:
                                           state.metric_outputs,
                                       )
 
-        # If batch mode is not generate, outputs will be logits
-        if state.batch['mode'] == 'generate':
+
+        if state.batch.get('mode') == 'generate':
             # Outputs are already detokenized
             logging_dict['outputs'] = state.outputs
+        elif isinstance(state.outputs, torch.Tensor):
+            # If batch mode is not generate, outputs will be logits
+            logging_dict['outputs'] = state.outputs.argmax(dim=-1)
 
         input_ids = state.batch['input_ids']
         logged_input = []
         assert state.dataloader is not None
+        dataset = state.dataloader.dataset  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
This makes things much more legible but also doubles the dataset in memory 😔. Most eval datasets are small and it wouldn't matter much, but we have run into problems when people using surprisingly large eval datasets before, so I don't think this is a good idea in the end.

Reply:
Is this a gc issue? Isn't the variable already defined in the `for input_list in` loop?

Reply:
Discussed offline, resolving -- should not result in a copy since it stores a pointer.